pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/gardeneruser/templates/scripts/reconcile.tpl.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash -eu
+
+DIR_SSH="/home/gardener/.ssh"
+PATH_AUTHORIZED_KEYS="$DIR_SSH/authorized_keys"
+PATH_SUDOERS="/etc/sudoers.d/99-gardener-user"
+USERNAME="gardener"
+
+# create user if missing
+id $USERNAME || useradd $USERNAME -mU
+
+# copy authorized_keys file
+mkdir -p $DIR_SSH
+cp -f "{{ .pathAuthorizedSSHKeys }}" $PATH_AUTHORIZED_KEYS
+chown $USERNAME:$USERNAME $PATH_AUTHORIZED_KEYS
+
+# remove unused legacy file
+if [ -f "{{ .pathPublicSSHKey }}" ]; then
+  rm -- "{{ .pathPublicSSHKey }}"
```

Comment:
I have never seen this syntax so far. Reading the man it is used when the filename starts with `-`, but here this is not the case. 
```suggestion
  rm -f "{{ .pathPublicSSHKey }}"
```